crux.el

```diff
@@ -456,23 +456,47 @@ See `file-attributes' for more info."
   (equal (crux-file-owner-uid filename)
          (user-uid)))
 
+(defun crux-already-root-p ()
+  (let ((remote-method (file-remote-p default-directory 'method))
+        (remote-user (file-remote-p default-directory 'user)))
+    (when (and remote-method
+               (or (member remote-method '("sudo" "su" "ksu" "doas"))
+                   (string= remote-user "root")))
+      (message "Already editing this file as root."))))
+
 (defun crux-find-alternate-file-as-root (filename)
   "Wraps `find-alternate-file' with opening FILENAME as root."
-  (find-alternate-file (concat "/sudo:root@localhost:" filename)))
+  (let ((remote-method (file-remote-p default-directory 'method))
+        (remote-user (file-remote-p default-directory 'user))
+        (remote-host (file-remote-p default-directory 'host))
+        (remote-localname (file-remote-p filename 'localname)))
+    (find-alternate-file (format "/%s:root@%s:%s"
+                                 (or remote-method "sudo")
+                                 (or remote-host "localhost")
+                                 (or remote-localname filename)))))
 
-(require 'ido)
 ;;;###autoload
 (defun crux-sudo-edit (&optional arg)
   "Edit currently visited file as root.
 
 With a prefix ARG prompt for a file to visit.
 Will also prompt for a file to visit if current
 buffer is not visiting a file."
-  (interactive "P")
+  (interactive)
```

Comment:
Seems to me you made a mistake removing this.

Reply:
Whoops! Agreed.